### ISSUE TITLE
[shelly] exception in discovery

### ISSUE THREAD

**Issue by J-N-K** [openhab/openhab-addons#8240]
```
2020-07-31 21:01:03.683 [ERROR] [y.mdns.internal.MDNSDiscoveryService] - Participant 'org.openhab.binding.shelly.internal.discovery.ShellyDiscoveryParticipant' threw an exception
java.lang.IllegalArgumentException: ID segment '960 series' contains invalid characters. Each segment of the ID must match the pattern [A-Za-z0-9_-]*.
	at org.eclipse.smarthome.core.common.AbstractUID.validateSegment(AbstractUID.java:97) ~[bundleFile:?]
	at org.eclipse.smarthome.core.common.AbstractUID.<init>(AbstractUID.java:75) ~[bundleFile:?]
	at org.eclipse.smarthome.core.common.AbstractUID.<init>(AbstractUID.java:58) ~[bundleFile:?]
	at org.eclipse.smarthome.core.thing.UID.<init>(UID.java:57) ~[bundleFile:?]
	at org.eclipse.smarthome.core.thing.ThingUID.<init>(ThingUID.java:43) ~[bundleFile:?]
	at org.openhab.binding.shelly.internal.discovery.ShellyThingCreator.getThingUID(ShellyThingCreator.java:32) ~[?:?]
	at org.openhab.binding.shelly.internal.discovery.ShellyDiscoveryParticipant.getThingUID(ShellyDiscoveryParticipant.java:205) ~[?:?]
	at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryService.serviceRemoved(MDNSDiscoveryService.java:241) [bundleFile:?]
	at javax.jmdns.impl.ListenerStatus$ServiceListenerStatus.serviceRemoved(ListenerStatus.java:81) [bundleFile:3.5.5]
	at javax.jmdns.impl.JmDNSImpl$5.run(JmDNSImpl.java:1322) [bundleFile:3.5.5]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_262]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_262]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_262]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_262]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_262]
```

The exception is from a log sent to me in private communication. A solution could be to use `UIDUtils.encode` in l. 31 of `ShellyThingCreator`. I'm a bit unsure about this since `960 series` does not look like a device id. MAybe someone with more knowledge about shelly @markus7017 can comment on that. Unfortunately the device causing this is unknown.

**Comment by markus7017** [comment]
Hi, there is no "960 series" Shelly device. So the discovery handler gets a request for a non-Shelly serviceName (startiung with "shelly"). I added some more checks to ensure that serviceName is != null, starts with "shelly" and also has a "-" (separator for the mac address). Will be included in the pending PR.

**Comment by Hilbrand** [comment]
Is this also fixed with #8198?

**Comment by markus7017** [comment]
yes and verified
